chia/server/address_manager_store.py

```diff
@@ -97,52 +78,105 @@ async def create_address_manager(cls, peers_file_path: Path) -> AddressManager:
-        """
-        Serialize the address manager's peer data to a file.
-        """
-        metadata: list[tuple[str, str]] = []
-        nodes: list[tuple[int, ExtendedPeerInfo]] = []
-        new_table_entries: list[tuple[int, int]] = []
+    async def serialize_bytes(cls, address_manager: AddressManager, peers_file_path: Path) -> None:
+        out = io.BytesIO()
+        nodes = io.BytesIO()
+        trieds = io.BytesIO()
+        new_table = io.BytesIO()
         unique_ids: dict[int, int] = {}
         count_ids: int = 0
 
-        log.info("Serializing peer data")
-        metadata.append(("key", str(address_manager.key)))
-
         for node_id, info in address_manager.map_info.items():
             unique_ids[node_id] = count_ids
             if info.ref_count > 0:
                 assert count_ids != address_manager.new_count
-                nodes.append((count_ids, info))
+                info.stream(nodes)
                 count_ids += 1
-        metadata.append(("new_count", str(count_ids)))
-
-        tried_ids = 0
-        for node_id, info in address_manager.map_info.items():
             if info.is_tried:
-                assert info is not None
-                assert tried_ids != address_manager.tried_count
-                nodes.append((count_ids, info))
-                count_ids += 1
-                tried_ids += 1
-        metadata.append(("tried_count", str(tried_ids)))
+                info.stream(nodes)
+
+        out.write(address_manager.key.to_bytes(32, byteorder="big"))
+        uint64(count_ids).stream(out)
 
+        count = 0
         for bucket in range(NEW_BUCKET_COUNT):
             for i in range(BUCKET_SIZE):
                 if address_manager.new_matrix[bucket][i] != -1:
-                    index = unique_ids[address_manager.new_matrix[bucket][i]]
-                    new_table_entries.append((index, bucket))
-
-        try:
-            # Ensure the parent directory exists
-            peers_file_path.parent.mkdir(parents=True, exist_ok=True)
-            start_time = timer()
-            await cls._write_peers(peers_file_path, metadata, nodes, new_table_entries)
-            log.debug(f"Serializing peer data took {timer() - start_time} seconds")
-        except Exception:
-            log.exception(f"Failed to write peer data to {peers_file_path}")
+                    count += 1
+                    uint64(unique_ids[address_manager.new_matrix[bucket][i]]).stream(new_table)
+                    uint64(bucket).stream(new_table)
+
+        # give ourselves a clue how long the new_table is
+        uint32(count).stream(out)
+        out.write(new_table.getvalue())
+
+        out.write(nodes.getvalue())
+        out.write(trieds.getvalue())
+        await write_file_async(peers_file_path, out.getvalue(), file_mode=0o644)
 
+    @classmethod
+    async def deserialize_bytes(cls, data: io.BytesIO) -> AddressManager:
```

Comment:
I don't think this needs to be `async` anymore either

Reply:
Done